web/app/view/BaseMap.js

```diff
@@ -82,26 +86,48 @@ Ext.define('Traccar.view.BaseMap', {
             view: this.mapView
         });
 
-        target = this.map.getTarget();
-        if (typeof target === 'string') {
-            target = Ext.get(target).dom;
-        }
+        target = this.map.getTargetElement();
+        this.map.on('pointermove', function (evt) {
+            target.style.cursor = this.hasFeatureAtPixel(evt.pixel) ? 'pointer' : '';
+        });
 
-        this.map.on('pointermove', function (e) {
-            var hit = this.forEachFeatureAtPixel(e.pixel, function (feature, layer) {
-                return true;
-            });
-            if (hit) {
-                target.style.cursor = 'pointer';
-            } else {
-                target.style.cursor = '';
-            }
+        popupElement = document.createElement("div");
+        popupElement.id = "popup";
+        document.body.appendChild(popupElement);
+
+        this.popupOverlay = new ol.Overlay({
+            element: popupElement,
+            autoPan: true,
+            autoPanAnimation: {
+                duration: 250
+            },
+            positioning: 'bottom-center',
+            stopEvent: false,
+            offset: [0, -150]
         });
+        this.map.addOverlay(this.popupOverlay);
+
+        that = this;
+        popupElement.onclick = function(evt) {
+            if (that.map.isMapClick) {
+                that.map.isPopupOverlayClick = true;
+            }
+        };
 
         this.map.on('click', function (e) {
-            this.map.forEachFeatureAtPixel(e.pixel, function (feature, layer) {
+            var feature = this.map.forEachFeatureAtPixel(e.pixel, function (feature, layer) {
                 this.fireEvent('selectFeature', feature);
+                return feature;
             }, this);
+
+            that.map.isMapClick = true;
+            window.setTimeout(function () {
+                if (!feature && !that.map.isPopupOverlayClick) {
+                    that.fireEvent('clickMap');
+                }
+                that.map.isPopupOverlayClick = false;
+                that.map.isMapClick = false;
+            }, 250);
```

Comment:
I wanted the popup to be closed when user click on the map (because the X button too small for mobile devices), but not when the user click a toolbar button (or anything else inside the popup area).
So, I had to determine if the click event on the map was inside the popup overlay or not. to do this I added popupElement.onclick listener.
The problem was that the click event inside the popup overlay occurs after the click event on the map, so I had to use a timer to sync the click test at the right time of the events.